json_encoder.go

```diff
@@ -0,0 +1,184 @@
+// +build !enable_binary_log
+
```

Comment:
I would have named the two encoders `encoder_json.go` and `encoder_cbor.go` so they stay grouped together.

Reply:
i can rename the files to encoder_json,  encoder_cbor.

Reply:
done